cmd/gateclient/client.go

```diff
@@ -368,6 +376,53 @@ func (m *GatewayClient) authenticateIAP() (string, error) {
 	return token, err
 }
 
+func (m *GatewayClient) authenticateGoogleServiceAccount() (err error) {
+	auth := m.Config.Auth
+	gsa := auth.GoogleServiceAccount
```

Comment:
Was following the pattern in the OAuth2 one, but I have no problem combining it. This does bring up a bug here though -- auth can be nil. So, probably just going to add an early out for that instead of combining.